### PR TITLE
feat: Rename trackedentityinstance DB table to trackedentity [ TECH-1594 ]

### PR DESCRIPTION
**Pull request by lucaCambi77**
see https://github.com/dhis2/dhis2-core/pull/14569

### CODE REVIEW

File: dhis-2/dhis-services/dhis-service-core/src/main/resources/org/hisp/dhis/program/hibernate/ProgramMessage.hbm.xml [hunk 1]
```diff
@@ -42,7 +42,7 @@
     <component name="recipients" class="org.hisp.dhis.program.message.ProgramMessageRecipients">
 
      <many-to-one name="trackedEntity" class="org.hisp.dhis.trackedentity.TrackedEntity"
-      column="trackedEntityInstanceId" foreign-key="fk_programmessage_trackedentityinstanceid" not-null="false" />
+      column="trackedEntityId" foreign-key="fk_programmessage_trackedentityinstanceid" not-null="false" />
```
Comment (teleivo) [inline dhis-2/dhis-services/dhis-service-core/src/main/resources/org/hisp/dhis/program/hibernate/ProgramMessage.hbm.xml:45]:
```suggestion
      column="trackedentityid" foreign-key="fk_programmessage_trackedentityinstanceid" not-null="false" />
```

as I think the columns are all lower case

File: dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/export/trackedentity/aggregates/DefaultTrackedEntityStore.java [hunk 2]
```diff
@@ -64,30 +64,30 @@ public class DefaultTrackedEntityStore extends AbstractStore implements TrackedE
...
           + "join program p on teop.programid = p.programid "
           + "join organisationunit o on teop.organisationunitid = o.organisationunitid "
-          + "join trackedentityinstance tei on teop.trackedentityinstanceid = tei.trackedentityinstanceid "
-          + "where teop.trackedentityinstanceid in (:ids)";
+          + "join trackedentity tei on teop.trackedentityId = tei.trackedentityid "
+          + "where teop.trackedentityId in (:ids)";
 
   private static final String GET_OWNERSHIP_DATA_FOR_TEIS_FOR_ALL_PROGRAM =
-      "SELECT tei.uid as tei_uid,tpo.trackedentityinstanceid, tpo.programid, tpo.organisationunitid, p.accesslevel,p.uid as pgm_uid "
+      "SELECT tei.uid as tei_uid,tpo.trackedentityId, tpo.programid, tpo.organisationunitid, p.accesslevel,p.uid as pgm_uid "
```
Comment (teleivo) [inline dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/export/trackedentity/aggregates/DefaultTrackedEntityStore.java:71]:
```suggestion
      "SELECT tei.uid as tei_uid,tpo.trackedentityid, tpo.programid, tpo.organisationunitid, p.accesslevel,p.uid as pgm_uid "
```

lowercase

File: dhis-2/dhis-support/dhis-support-db-migration/src/main/resources/org/hisp/dhis/db/migration/2.41/V2_41_24__TrackedEntityInstanceToTrackedEntity.sql [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+begin
+  if exists(select 1
+    from information_schema.columns
+    where table_schema != 'information_schema' and table_name='trackedentityaudit' and column_name='trackedentityinstanceauditid')
+  then
+alter table trackedentityaudit rename column trackedentityinstanceauditid to trackedentityauditid;
+end if;
+end $$;
+
+do $$
```
Comment (teleivo) [inline dhis-2/dhis-support/dhis-support-db-migration/src/main/resources/org/hisp/dhis/db/migration/2.41/V2_41_24__TrackedEntityInstanceToTrackedEntity.sql:38]: it looks like this block is duplicated but maybe I am mistaken 🤔

**Review by teleivo — COMMENTED** [review]
Nice 😄 
One small question on whether a portion of the migration is duplicated.

File: dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/export/trackedentity/aggregates/DefaultTrackedEntityStore.java (hunk 2 above)
Comment (lucaCambi77) [reply dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/export/trackedentity/aggregates/DefaultTrackedEntityStore.java]: thanks!

**Review by teleivo — APPROVED** [review]

**Review by maikelarabori — APPROVED** [review]

**Review by gnespolino — COMMENTED** [review]
LGTM

**Pull request merged**